### ISSUE TITLE
macOS packages installs a deprecated service

### ISSUE THREAD

**Issue by DFolchA** [wazuh/wazuh-packages#2206]
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.4.2| Agent | Install | macOS |

The service installed by the agent package in macOS systems is deprecated, we need to change it so the package installs the new service system.

The installer needs to detect the service system used in each macOS version and install the correct service.

### Tasks 

- [ ] Investigate services in new macOS versions.
- [ ] Modify SPECS to install the correct service for each macOS version.

### Validation

- [ ] The package installs the correct service for each system.
- [ ] The services run properly.

**Comment by c-bordon** [comment]
## Update report

I was doing research and I could not find information about the deprecated services and I am not finding evidence to support it, I would need more information on which versions it is failing

**Comment by c-bordon** [comment]
I was carrying out tests in different systems and investigating about the management of services with launchctl and I cannot identify from which version it stopped working:

https://web.archive.org/web/20180218090359/https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/launchctl.1.html

```console
sh-3.2# installer -pkg wazuh-agent-4.4.2-1.pkg -target /
installer: Package name is Wazuh Agent
installer: Installing at base path /
installer: The install was successful.
sh-3.2# launchctl list wazuh-agent
Could not find service "wazuh-agent" in domain for system
sh-3.2# /bin/launchctl load /Library/LaunchDaemons/com.wazuh.agent.plist
/Library/LaunchDaemons/com.wazuh.agent.plist: service already loaded
sh-3.2# /bin/launchctl list wazuh-agent
Could not find service "wazuh-agent" in domain for system
sh-3.2# /bin/launchctl list /Library/LaunchDaemons/com.wazuh.agent.plist
Could not find service "/Library/LaunchDaemons/com.wazuh.agent.plist" in domain for system

```

**Comment by c-bordon** [comment]
I was carrying out new tests on some versions of macOS and I have found that launchctl works correctly, I will continue investigating on this:

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "TimeOut" = 30;
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 2668;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
sh-3.2# sw_vers
ProductName:    Mac OS X
ProductVersion: 10.12.6
BuildVersion:   16G29


sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "TimeOut" = 30;
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
sh-3.2# sw_vers
ProductName:    Mac OS X
ProductVersion: 10.13.6
BuildVersion:   17G65

```

**Comment by c-bordon** [comment]
## macOS Mojave

### launchctl start

```console
sh-3.2# launchctl start com.wazuh.agent
```

### launchctl print 

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 1
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = running

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 3
        successive crashes = 0
        pid = 1486
        immediate reason = one-shot
        forks = 3
        execs = 1
        initialized = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                supports idle hysteresis = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl stop

```console
sh-3.2# launchctl stop com.wazuh.agent
```

### launchctl print

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 0
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = waiting

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                supports idle hysteresis = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl list

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "TimeOut" = 30;
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 1486;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
```

**Comment by c-bordon** [comment]
## macOS Sierra

### launchctl start

```console
sh-3.2# launchctl start com.wazuh.agent
```

### launchctl print 

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 1
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = running

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        excessive crashing = 0
        pid = 3318
        immediate reason = one-shot
        forks = 6
        execs = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                dirty at shutdown = 0
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                requires reap = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl stop

```console
sh-3.2# launchctl stop com.wazuh.agent
```

### launchctl print

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 0
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = waiting

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        excessive crashing = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                dirty at shutdown = 0
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                requires reap = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl list

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "TimeOut" = 30;
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 3413;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
```

### launchctl version

```console
sh-3.2# launchctl version
Darwin Bootstrapper Version 4.0.0: Sun May  7 19:34:52 PDT 2017; root:libxpc_executables-972.70.1~4/launchd/RELEASE_X86_64

```

**Comment by c-bordon** [comment]
## macOS High Sierra

### launchctl start

```console
sh-3.2# launchctl start com.wazuh.agent
```

### launchctl print 

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 1
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = running

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        excessive crashing = 0
        pid = 858
        immediate reason = one-shot
        forks = 5
        execs = 1
        initialized = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                dirty at shutdown = 0
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl stop

```console
sh-3.2# launchctl stop com.wazuh.agent
```

### launchctl print

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 0
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = waiting

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        excessive crashing = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                dirty at shutdown = 0
                low priority i/o = 0
                low priority background i/o = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 0
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                disallow all lookups = 0
                system service = 0
        }
}
```

### launchctl list

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "TimeOut" = 30;
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 958;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
```

### launchctl version

```console
sh-3.2# launchctl version
Darwin Bootstrapper Version 5.3.0: Thu May 31 18:56:58 PDT 2018; root:libxpc_executables-1205.70.9~7/launchd/RELEASE_X86_64
```

**Comment by c-bordon** [comment]
## macOS Catalina

### launchctl start

```console
sh-3.2# launchctl start com.wazuh.agent
```

### launchctl print 

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 1
        copy count = 0
        one shot = 0
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = running

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        pid = 1061
        immediate reason = one-shot
        forks = 5
        execs = 2
        initialized = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        instances = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                low priority i/o = 0
                low priority background i/o = 0
                dataless file mode = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                supports idle hysteresis = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 1
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                is copy = 0
                disallow all lookups = 0
                system service = 0
                protected by submitter = 0
        }
}
```

### launchctl stop

```console
sh-3.2# launchctl stop com.wazuh.agent
```

### launchctl print

```console
sh-3.2# launchctl print system/com.wazuh.agent
com.wazuh.agent = {
        active count = 0
        copy count = 0
        one shot = 0
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = waiting

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = com.apple.xpc.launchd.domain.system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        successive crashes = 0
        last exit code = 0

        event triggers = {
        }

        endpoints = {
        }

        dynamic endpoints = {
        }

        pid-local endpoints = {
        }

        instance-specific endpoints = {
        }

        event channels = {
        }

        sockets = {
        }

        instances = {
        }

        spawn type = daemon
        spawn role = (null)
        jetsam priority = 3
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = {
                partial import = 0
                launchd bundle = 0
                xpc bundle = 0
                keepalive = 0
                runatload = 1
                low priority i/o = 0
                low priority background i/o = 0
                dataless file mode = 0
                legacy timer behavior = 0
                exception handler = 0
                multiple instances = 0
                supports transactions = 0
                supports pressured exit = 0
                supports idle hysteresis = 0
                enter kdp before kill = 0
                wait for debugger = 0
                app = 0
                system app = 0
                creates session = 0
                inetd-compatible = 0
                inetd listener = 0
                abandon process group = 0
                one-shot = 0
                event monitor = 0
                penalty box = 0
                pended non-demand spawn = 0
                role account = 0
                launch only once = 0
                system support = 0
                app-like = 0
                inferred program = 1
                joins gui session = 0
                joins host session = 0
                parameterized sandbox = 1
                resolve program = 0
                abandon coalition = 0
                high bits aslr = 0
                extension = 0
                nano allocator = 0
                no initgroups = 0
                start on fs mount = 0
                endpoints initialized = 1
                is copy = 0
                disallow all lookups = 0
                system service = 0
                protected by submitter = 0
        }
}
```

### launchctl list

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 1217;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
```

### launchctl version

```console
sh-3.2# launchctl version
Darwin Bootstrapper Version 7.0.0: Thu Sep 26 18:17:21 PDT 2019; root:libxpc_executables-1738.40.10~19/launchd/RELEASE_X86_64
```

**Comment by c-bordon** [comment]
## macOS Monterrey

### launchctl start

```console
sh-3.2# launchctl start com.wazuh.agent
```

### launchctl print 

```console
sh-3.2# launchctl print system/com.wazuh.agent
system/com.wazuh.agent = {
        active count = 1
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = running

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        pid = 1174
        immediate reason = one-shot
        forks = 5
        execs = 2
        initialized = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        spawn type = daemon (3)
        jetsam priority = 4
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = runatload | inferred program
}

```

### launchctl stop

```console
sh-3.2# launchctl stop com.wazuh.agent
```

### launchctl print

```console
sh-3.2# launchctl print system/com.wazuh.agent
system/com.wazuh.agent = {
        active count = 1
        path = /Library/LaunchDaemons/com.wazuh.agent.plist
        state = SIGTERMed

        program = /Library/StartupItems/WAZUH/Wazuh-launcher
        arguments = {
                /Library/StartupItems/WAZUH/Wazuh-launcher
        }

        default environment = {
                PATH => /usr/bin:/bin:/usr/sbin:/sbin
        }

        environment = {
                XPC_SERVICE_NAME => com.wazuh.agent
        }

        domain = system
        minimum runtime = 10
        exit timeout = 5
        runs = 2
        pid = 1174
        immediate reason = one-shot
        forks = 12
        execs = 2
        initialized = 1
        trampolined = 1
        started suspended = 0
        proxy started suspended = 0
        last exit code = 0

        spawn type = daemon (3)
        jetsam priority = 4
        jetsam memory limit (active) = (unlimited)
        jetsam memory limit (inactive) = (unlimited)
        jetsamproperties category = daemon
        submitted job. ignore execute allowed
        jetsam thread limit = 32
        cpumon = default

        properties = runatload | inferred program
}
```

### launchctl list

```console
sh-3.2# launchctl list com.wazuh.agent
{
        "LimitLoadToSessionType" = "System";
        "Label" = "com.wazuh.agent";
        "OnDemand" = true;
        "LastExitStatus" = 0;
        "PID" = 1284;
        "Program" = "/Library/StartupItems/WAZUH/Wazuh-launcher";
        "ProgramArguments" = (
                "/Library/StartupItems/WAZUH/Wazuh-launcher";
        );
};
```

### launchctl version

```console
sh-3.2# launchctl version
Darwin Bootstrapper Version 7.0.0: Fri Oct  1 22:13:44 PDT 2021; root:libxpc_executables-2235.40.11~145/launchd/RELEASE_X86_64
```

**Comment by mjcr99** [comment]
## Update
**(04/07/2024)** Reviewing current report and reading information about how to detect a deprecated service in macOS.

**(08/07/2024)** I have created this [testing PR](https://github.com/wazuh/wazuh/pull/24482) to test the service installation status using GHA machines in order to test the behavior with the newest Mac OS versions in a fast way.
The run workflow is the following: https://github.com/wazuh/wazuh/actions/runs/9837269563

As it can be seen, no references to deprecated service are found while checking the service status.

**Comment by MarcelKemp** [comment]
The problem is not replicated in the supported versions, so I proceed to close the issue.
